### ISSUE TITLE
external projects don't yet have an automated way to populate them.

### ISSUE THREAD

**Issue by robertosfield** [SaschaWillems/VulkanInfiniteProceduralTerrain#2]
When I run cmake . in the VulkanInfiniteProceduralTerrain directory on my Kubuntu 22.04 system I get an error:

CMake Error at CMakeLists.txt:118 (add_subdirectory):
  add_subdirectory given source "external" which is not an existing
  directory.

I looked for them to be pulled in as submodules so tried a:

   git submodule init
   git submodule update

But this didn't pull anything in.  I can't see any entry for modules in the .git setup so I presume this simply hasn't been setup/checked in yet.

--

The full CMake output is:

~/3rdParty/VulkanInfiniteProceduralTerrain$ cmake .
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using module to find Vulkan
-- Found Vulkan: /home/robert/3rdParty/VulkanSDK/1.3.224.0/x86_64/lib/libvulkan.so  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found PkgConfig: /bin/pkg-config (found version "0.29.2") 
-- Checking for module 'xcb'
--   Found xcb, version 1.14
-- /home/robert/3rdParty/VulkanSDK/1.3.224.0/x86_64/lib/libvulkan.so
CMake Error at CMakeLists.txt:118 (add_subdirectory):
  add_subdirectory given source "external" which is not an existing
  directory.

-- Configuring incomplete, errors occurred!

**Comment by SaschaWillems** [comment]
Yeah, this is still pretty much a work-in-progress and can't be compiled out of the box. I'll try to add everything required to build this within the coming weeks.

**Comment by SaschaWillems** [comment]
I have added all required external files and also a base set of assets required to run this. Can you check if this will compile and run now for you?

**Comment by robertosfield** [comment]
Thanks I've pulled your changes into my fork and it's now completed the cmake . and has progressed onto building the source.  I'm getting build errors but it's late here leave it for another day to see what need adapting for Linux build.

**Comment by SaschaWillems** [comment]
Thanks :)

It's very much possible that I broke things on Linux, as I'm currently only building on Windows. Feel free to write down any Linux build errors in here so I can fix them.

**Comment by robertosfield** [comment]
On Wed, 7 Sept 2022 at 19:15, Sascha Willems ***@***.***>
wrote:

> Thanks :)
>
> It's very much possible that I broke things on Linux, as I'm currently
> only building on Windows. Feel free to write down any Linux build errors in
> here so I can fix them.
>
> I will provide error reports/fixes once I get time.

I broke my arm so my productivity is rather low right now.... I'll have
this cast on for 3 more weeks.

FYI, longer term I'd like to build a VulkanSceneGraph demo that touches
upon many of the elements you are tackling in your InfinitProcudralTerrain
program. I still have VSG-1.0 to complete so it'll have to wait till after
that.

In particular I'm curious about the possibility of mixing procedural
terrain/trees etc. with real wor;d imageryDEMS data with real-world
cultural elements such as roads/buildings/runways etc.  Years of work....

**Comment by SaschaWillems** [comment]
> I broke my arm so my productivity is rather low right now.... I'll have
this cast on for 3 more weeks.

That's sad to hear. A programmer's nightmare. Hope you'll recover soon.

I may give it a shot on Linux myself and see what's broken in the meantime.
